### PR TITLE
First part User Module

### PR DESCRIPTION
**Pull request by raeeschachar**

### CODE REVIEW

File: contacts/models.py [hunk 1]
```diff
@@ -8,6 +9,7 @@ class Contact(models.Model):
             message="Phone number must be entered in the format: '+999999999'. Up to 20 digits allowed."
     )
 
+    user = models.ForeignKey(User, null=True, blank=True)
```
Comment (MaryKhan) [inline contacts/models.py]: this is fine, but i wud like to have a separate app for user with its model having fields:
first_name, last_name, email, phone, address etc. but lets do it in separate branch after this gets merged!

secondly, a contact should never be created without a user.

File: contacts/views.py [hunk 2]
```diff
@@ -82,3 +85,45 @@ def post(self, request, pk):
...
+        if form.is_valid():
+            un = form.cleaned_data['username']
+            pw = form.cleaned_data['password']
+            user = authenticate(username=un, password=pw)
+            if user:
+                if user.is_active:
+                    login(request, user)
+                    return HttpResponseRedirect(reverse('contacts:contact'))
+                else:
+                    print("Invalid account")
```
Comment (MaryKhan) [inline contacts/views.py]: remove print statements and return form with errors.

File: contacts/views.py [hunk 3]
```diff
@@ -82,3 +85,45 @@ def post(self, request, pk):
         contact = Contact.objects.get(id=pk)
         contact.delete()
         return HttpResponseRedirect(reverse('contacts:contact'))
+
+
+class LoginView(View):
```
Comment (MaryKhan) [inline contacts/views.py:105]: all these views belong to user not contact.
create a separate app for users and move these all over there!

File: contacts/views.py [hunk 4]
```diff
@@ -82,3 +85,45 @@ def post(self, request, pk):
...
+    template_name = 'contacts/home.html'
+
+    def get(self, request):
+        return render(request, self.template_name)
+
+
+class LogoutView(View):
+    template_name = 'contacts/home.html'
+
+    def get(self, request):
```
Comment (MaryKhan) [inline contacts/views.py:139]: is it working? we're not logging user out here!

File: contacts/urls.py [hunk 5]
```diff
@@ -5,13 +5,16 @@
...
     url(r'^(?P<pk>[0-9]+)/?$', views.ContactDetailView.as_view(), name='detail'),
     url(r'^add_contact/?$', views.AddContactView.as_view(), name='add_contact'),
     url(r'^add_address/?$', views.AddContactAddressView.as_view(), name='add_address'),
     url(r'^update_contact/(?P<pk>[0-9]+)/?$', views.UpdateContactView.as_view(), name='update_contact'),
     url(r'^update_contact_address/(?P<pk>[0-9]+)/?$',
         views.UpdateContactAddressView.as_view(), name='update_contact_address'
         ),
-    url(r'^delete/(?P<pk>\d+)/?$', views.DeleteContactView.as_view(), name='delete_contact')
+    url(r'^delete/(?P<pk>\d+)/?$', views.DeleteContactView.as_view(), name='delete_contact'),
+    url(r'^login/?$', views.LoginView.as_view(), name='user_login'),
```
Comment (MaryKhan) [inline contacts/urls.py]: why these names are inconsistent? either add `user_` with logout or remove it from login too.

File: contacts/migrations/0003_contact_user.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (MaryKhan) [inline contacts/migrations/0003_contact_user.py]: merge these migrations into one!

File: contacts/templates/contacts/login.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<h1>Login</h1>
+{% block content %}
+    {% if error_message %}<p><strong>{{ error_message }}</strong></p>{% endif %}
+    {% include 'contacts/form.html' with form=form %}
+
+{% endblock %}
```
Comment (MaryKhan) [inline contacts/templates/contacts/login.html]: pep8: extra line missing.

**Review by MaryKhan — CHANGES_REQUESTED** [review]

File: contacts/templates/contacts/login.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<h1>Login</h1>
+{% block content %}
+    {% if error_message %}<p><strong>{{ error_message }}</strong></p>{% endif %}
```
Comment (MaryKhan) [inline contacts/templates/contacts/login.html:3]: where are these `error_message` coming from?

**Review by MaryKhan — CHANGES_REQUESTED** [review]

File: contacts/urls.py (hunk 5 above)
Comment (raeeschachar) [reply contacts/urls.py]: Done

File: contacts/templates/contacts/login.html (hunk 7 above)
Comment (raeeschachar) [reply contacts/templates/contacts/login.html]: Done

File: contacts/views.py (hunk 4 above)
Comment (raeeschachar) [reply contacts/views.py:139]: Done

File: contacts/templates/contacts/login.html (hunk 8 above)
Comment (raeeschachar) [reply contacts/templates/contacts/login.html:3]: Done

File: contacts/views.py (hunk 2 above)
Comment (raeeschachar) [reply contacts/views.py]: Done

**Comment by raeeschachar** [comment]
@MaryKhan 
There are still a couple of bugs, on adding address; all contacts appear instead of user specific ones. Working on that.

File: contacts/templates/contacts/contact_list.html [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
 
 {% block content %}
-<h1>My Contacts</h1>
+<h3>My Contacts</h3>
 
     <p><a href="{% url 'contacts:add_contact' %}">Add Contact</a></p>
     <p><a href="{% url 'contacts:add_address' %}">Add Address Against a Contact</a></p>
     <br>
-    {% for contact in my_contacts %}
+    {% for contact in specific %}
```
Comment (MaryKhan) [inline contacts/templates/contacts/contact_list.html]: `for contact in specific` doesn't make any sense to me.
list name should reflect the data it contains.

**Review by MaryKhan — CHANGES_REQUESTED** [review]

**Pull request merged**